src/main/scala/mesosphere/marathon/MarathonConf.scala

```diff
@@ -307,5 +307,12 @@ trait MarathonConf
     noshort = true,
     hidden = true,
     default = Some(MesosHeartbeatMonitor.DEFAULT_HEARTBEAT_FAILURE_THRESHOLD))
+
+  lazy val drainingTime = opt[Long](
+    "draining_time",
```

Comment:
I was a little unsure about the idea of the drainingTime. To me it feels like Marathon should decline offers as soon as a maintenance window is scheduled. The more I think about it the more sense it makes though. I guess I can set drainingTime to the same amount of time we set the start of the maintenance window in the future (or slightly longer). That way it will start declining the offers as soon as we schedule the maintenance window.